packages/program/src/functions/multicall-scope.ts

```diff
@@ -20,7 +21,9 @@ export class MultiCallInvocationScope<TReturn = any> extends BaseInvocationScope
   constructor(contract: AbstractContract, funcScopes: Array<FunctionInvocationScope>) {
     super(contract, true);
     this.addCalls(funcScopes);
-    this.validateHeapTypeReturnCalls();
+    if (this.program.interface.jsonAbi.encoding !== ENCODING_V1) {
```

Comment:
`v1` can handle multiple heaps in a single multicall, so we should not validate that here.